recipe/recipe.yaml

```diff
@@ -0,0 +1,221 @@
+context:
+  version: 1.3.1
+  build_num: 2
+
+recipe:
+  name: zlib-split
+  version: ${{ version }}
+
+source:
+  url:
+    - http://zlib.net/zlib-${{ version }}.tar.gz
+    - https://gnupg.org/ftp/gcrypt/zlib/zlib-${{ version }}.tar.gz
+  sha256: 9a93b2b7dfdac77ceba5a558a580e74667dd6fede4585b91eefb60f03b72df23
+
+build:
+  number: ${{ build_num }}
+
+cache:
+  requirements:
+    build:
+      - cmake
+      - ninja
+      - if: cross_target_platform == "emscripten-wasm32"
+        then:
+          - emscripten
+        else:
+          - ${{ compiler('c') }}
+          - ${{ stdlib('c') }}
+  build:
+    script:
+      # rattler-build figures out the {.sh,.bat} extensions
+      file: build_global
+      # see https://github.com/prefix-dev/rattler-build/issues/1056
+      env:
+        CROSS_TARGET_PLATFORM: ${{ cross_target_platform }}
```

Comment:
this should not be necessary since you are using the `cross_target_platform` variable in the cache (to determine the requirements). In that sense, you might have found a bug in rattler-build, indeed!

Reply:
That was what I was trying to communicate in https://github.com/prefix-dev/rattler-build/issues/1056 😅

Reply:
Well you never said that you are using them in the recipe.yaml output. You only showed me bits and pieces from the build script :D

Reply:
> Well you never said that you are using them in the recipe.yaml output

There are [two](https://github.com/prefix-dev/rattler-build/issues/1056#issuecomment-2340446481) [comments](https://github.com/prefix-dev/rattler-build/issues/1056#issuecomment-2340475603) where I'm talking about nothing else than variables picked up from the recipe (rather than the build script).